### PR TITLE
[doc] Adding PR guidelines

### PR DESCRIPTION
**Pull request by wrwg**
The file `./third_party/move/documentation/coding_guidelines.md` already contains useful guidelines for coding. This PR adds guidelines for code review.

### CODE REVIEW

**Review by vgao1996 — APPROVED** [review]

File: third_party/move/documentation/coding_guidelines.md [hunk 1]
```diff
@@ -23,6 +23,29 @@ $ cargo xclippy --all-targets
 
 In general, we follow the recommendations from [rust-lang-nursery](https://rust-lang.github.io/api-guidelines/) and [The Rust Programming Language](https://doc.rust-lang.org/book/).  The remainder of this guide provides detailed guidelines on specific topics in order to achieve uniformity of the codebase.
 
+## Code review
+
+We require every PR to have approval by at least two reviewers. This is enforced by the CI presubmit.
+
+### Guidelines for Authors of PRs
+
+- Have a meaningful description of the PR
```
Comment (brmataptos) [inline third_party/move/documentation/coding_guidelines.md:32]: Link to fixed issues.

When merging the PR as as commit(s), be sure to include
- purpose of the change
- references to fixed issues and design documents

Remember that future code maintainers will use your commit messages to understand why the code was changed and why it is written the way it is.

Comment (brmataptos) [reply third_party/move/documentation/coding_guidelines.md:32]: Btw, these were suggested text additions.  Not comments to you.  Sorry for the lack of framing.

File: third_party/move/documentation/coding_guidelines.md [hunk 2]
```diff
@@ -23,6 +23,29 @@ $ cargo xclippy --all-targets
...
+    - the purpose of a PR
+    - major design decisions
+    - testing strategy
```
Comment (vineethk) [inline third_party/move/documentation/coding_guidelines.md:35]: Additionally, a "security argument/strategy"?

**Review by vineethk — APPROVED** [review]
LGTM.

File: third_party/move/documentation/coding_guidelines.md [hunk 3]
```diff
@@ -23,6 +23,29 @@ $ cargo xclippy --all-targets
...
+    - any additional, non-primary changes related to the PR. It is OK to do minor refactorings of the code as you visit it.
+- Respond to each comment of the reviewers. If you think you resolved a request for a change, just use 'done' as a response. If you diasgree with the reviewer, explain why
+- Once you are done with addressing comments, indicate this to the reviewers by adding a top-level comment to the PR. You can use 'PTLA' ('please take a look again') as an acronym for this purpose.
+- If possible, avoid force push so the reviewer can see how you changed code in comparison. There are exceptions to this, for example if you need to rebase
```
Comment (aalok-t) [inline third_party/move/documentation/coding_guidelines.md:39]: Should we add any specifics about individual commits? AWS had strict rules about how each significant change should be a separate commit, and also certain rules and guidelines around rebasing.

File: third_party/move/documentation/coding_guidelines.md [hunk 4]
```diff
@@ -23,6 +23,29 @@ $ cargo xclippy --all-targets
...
+We require every PR to have approval by at least two reviewers. This is enforced by the CI presubmit.
+
+### Guidelines for Authors of PRs
+
+- Have a meaningful description of the PR
+    - the purpose of a PR
+    - major design decisions
+    - testing strategy
+    - any additional, non-primary changes related to the PR. It is OK to do minor refactorings of the code as you visit it.
+- Respond to each comment of the reviewers. If you think you resolved a request for a change, just use 'done' as a response. If you diasgree with the reviewer, explain why
```
Comment (georgemitenkov) [inline third_party/move/documentation/coding_guidelines.md:37]: diasgree--> disagree

Comment (georgemitenkov) [reply third_party/move/documentation/coding_guidelines.md:37]: nit: missing . in the end of the sentence?

File: third_party/move/documentation/coding_guidelines.md (hunk 3 above)
Comment (georgemitenkov) [inline third_party/move/documentation/coding_guidelines.md:39]: nit: no dot in the end of a sentence?

**Review by georgemitenkov — APPROVED** [review]

**Review by davidiw — COMMENTED** [review]
btw we do have this: https://aptos.dev/community/rust-coding-guidelines

File: third_party/move/documentation/coding_guidelines.md (hunk 1 above)
Comment (wrwg) [reply third_party/move/documentation/coding_guidelines.md:32]: Done

File: third_party/move/documentation/coding_guidelines.md (hunk 2 above)
Comment (wrwg) [reply third_party/move/documentation/coding_guidelines.md:35]: Done

File: third_party/move/documentation/coding_guidelines.md (hunk 3 above)
Comment (wrwg) [reply third_party/move/documentation/coding_guidelines.md:39]: We don't have such strict policies intentionally (we are a startup not a big corp and pace is important -- though many of us come from Meta/Google with similar policies), but there is a point to not mixing totally independent things. I added: "Avoid merging independent things into one PR. (Except minor refactoring and bug fixes as mentioned above.)"

File: third_party/move/documentation/coding_guidelines.md (hunk 4 above)
Comment (wrwg) [reply third_party/move/documentation/coding_guidelines.md]: Done

**Comment by wrwg** [comment]
> btw we do have this: https://aptos.dev/community/rust-coding-guidelines

This one here is more exhaustive, but there is also some overlap. I guess I need to look at unifying those ... not sure when I have time for this.

**Comment by wrwg** [comment]
> > btw we do have this: https://aptos.dev/community/rust-coding-guidelines
> 
> This one here is more exhaustive, but there is also some overlap. I guess I need to look at unifying those ... not sure when I have time for this.

Perhaps just pull out the PR guidelines into an independent file? Because that is what I'm caring here about, it was an action item of the VM summit were PRs were identified as a weak spot, with low quality both on submitters and reviewer end.

**Comment by wrwg** [comment]
> > > btw we do have this: https://aptos.dev/community/rust-coding-guidelines
> > 
> > 
> > This one here is more exhaustive, but there is also some overlap. I guess I need to look at unifying those ... not sure when I have time for this.
> 
> Perhaps just pull out the PR guidelines into an independent file? Because that is what I'm caring here about, it was an action item of the VM summit were PRs were identified as a weak spot, with low quality both on submitters and reviewer end.

Going to merge this now as is -- this is also part of public Move repo (but not the aptos.dev site), it makes sense to seperate. However, we should try to align both. The added PR guidelines would be for Move work for now only.
